wrapanapi/systems/google.py

```diff
@@ -581,12 +581,14 @@ def get_vm(self, name, zone=None, try_all_zones=False):
     def create_vm(self):
         raise NotImplementedError
 
-    def list_templates(self, include_public=False):
+    def list_templates(self, include_public=False, public_projects=None):
         images = self._compute.images()
         results = []
         projects = [self._project]
         if include_public:
             projects.extend(IMAGE_PROJECTS)
+            if public_projects:
+                projects.extend(public_projects)
```

Comment:
```suggestion
if include_public:
  if project_list:
      projects.extend(public_projects)
  else:
      projects.extend(IMAGE_PROJECTS)
```

Reply:
I think this looks more appropriate if there is an ask for public_projects then only extend otherwise go with the IMAGE_PROJECTS as usual.

Reply:
We took 5 lines that we could have done in 4 :)

This change should have been done because your `projects.extend(IMAGE_PROJECTS)` not depend on the `if include_public` condition. 

If project_list is given then, you adding to list of images that should also have  `IMAGE_PROJECTS `.

Is that the right expectation? OR is `public_projects  OR IMAGE_PROJECTS` situation ?

Reply:
I feel this looks more appropriate for instance what if the cloud provider is not in the IMAGE_PROJECTS list and we are passing it explicitly like we are doing in our test for rhel-sap-cloud , we only need those images list that we pass not all from the IMAGE_PROJECTS list.

Another part of it is if we want all the images, anyway the public_project is none that will take care of providing all the images as before.

Reply:
So what I get is its `IMAGE_PROJECTS` or `public_projects`